internal/metrics/service_monitor.go

```diff
@@ -42,12 +42,17 @@ func NewServiceMonitor(namespace, name, portName string, owner metav1.OwnerRefer
 		Endpoints: endpoint,
 		Selector: metav1.LabelSelector{
 			MatchLabels: map[string]string{
-				"logging-infra": "support",
+				"logging-infra":            "support",
+				constants.LabelK8sInstance: name,
```

Comment:
I wonder if we need another selector of similarity to the "instance" to further restrict it?  Maybe the comonent (e.g. logfilemetricexporter)?